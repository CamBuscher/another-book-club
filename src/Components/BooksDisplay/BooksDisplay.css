.bookshelf {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

.bookPreview {
  border: 1px solid black;
  margin: 10px;
  width: 22%;
  min-width: 250px;
  max-width: 400px;
  box-shadow: 3px 3px 8px black;
  padding: 5px;
}

.bookPreview img {
  width: 65%;
  max-width: 120px;
}

.bookPreview h4, h5 {
  color: whitesmoke;
  text-shadow: 2px 2px 4px black;
}

