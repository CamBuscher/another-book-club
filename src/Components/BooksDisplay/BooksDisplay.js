import React from 'react';

export const BooksDisplay = ({books}) => {
  const bookPreviews = books.map(book => {
    let previewImage;

    !book.imageLinks ? 
      previewImage = 'https://www.freeiconspng.com/uploads/no-image-icon-23.jpg' :
      previewImage = book.imageLinks.smallThumbnail;

    let author ;
    
    !book.authors ? 
      author = 'Author not available' :
      author = book.authors[0];

    return (
      <div key={book.publishedDate}>
        <h4>{book.title}</h4>
        <h5>Author: {author}</h5>
        <img className='bookPreviewImg' src={previewImage} />
      </div>
    )
  });

  return (
    <div>
      {bookPreviews}
    </div>
  );
};