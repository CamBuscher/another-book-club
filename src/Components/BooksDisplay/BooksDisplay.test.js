import React from 'react';
import { BooksDisplay } from './BooksDisplay';
import { shallow, mount } from 'enzyme';

describe('BooksDisplay', () => {
  let wrapper
  let mockprops

  it('should match snapshot when given books', () => {
    mockprops = {
      books: [{
        title: 'Annihilation',
        authors: ['Jeff'],
        imageLinks: {
          thumbnail: 'book.jpg'
        }
      }],
      currentClub: {
        books: []
      },
      addBookToClub: jest.fn()
    };
    wrapper = shallow(<BooksDisplay {...mockprops} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when given no books', () => {
    mockprops = {
      currentClub: {
        books: []
      },
      addBookToClub: jest.fn()
    };
    wrapper = shallow(<BooksDisplay {...mockprops} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should be an empty div if no books given', () => {
    mockprops = {books: []};
    wrapper = shallow(<BooksDisplay {...mockprops}/>);

    const bookPreviews = wrapper.find('.bookPreview');

    expect(bookPreviews).toHaveLength(0)
  });

  it('should have a bookPreview div if books received', () => {
    mockprops = {
      books: [{
        title: 'Annihilation',
        authors: ['Jeff'],
        imageLinks: {
          thumbnail: 'book.jpg'
        }
      }],
      currentClub: {
        books: []
      },
      addBookToClub: jest.fn()
    };
    wrapper = shallow(<BooksDisplay {...mockprops} />);

    const bookPreviews = wrapper.find('.bookPreview');

    expect(bookPreviews).toHaveLength(1);
  });

  it('should show a no-image icon if given book has no image', () => {
    mockprops = {
      books: [{
        title: 'Annihilation',
        authors: ['Jeff']
      }],
      currentClub: {
        books: []
      },
      addBookToClub: jest.fn()
    };
    wrapper = shallow(<BooksDisplay {...mockprops} />);

    const image = wrapper.find('.bookPreview').find('img');

    expect(image.props().src).toEqual('https://www.freeiconspng.com/uploads/no-image-icon-23.jpg');
  });

  it('should show image link if given a book image', () => {
    mockprops = {
      books: [{
        title: 'Annihilation',
        authors: ['Jeff'],
        imageLinks: {
          thumbnail: 'book.jpg'
        }
      }],
      currentClub: {
        books: []
      },
      addBookToClub: jest.fn()
    };
    wrapper = shallow(<BooksDisplay {...mockprops} />);

    const image = wrapper.find('.bookPreview').find('img');

    expect(image.props().src).toEqual('book.jpg');
  });

  it('should show no author available if not given an author', () => {
    mockprops = {
      books: [{
        title: 'Annihilation',
        imageLinks: {
          thumbnail: 'book.jpg'
        }
      }],
      currentClub: {
        books: []
      },
      addBookToClub: jest.fn()
    };
    wrapper = shallow(<BooksDisplay {...mockprops} />);

    const author = wrapper.find('.bookPreview').find('h5');

    expect(author.text()).toEqual('Author: Author not available');
  });

  it('should show author name if given an author', () => {
    mockprops = {
      books: [{
        title: 'Annihilation',
        authors: ['jeff'],
        imageLinks: {
          thumbnail: 'book.jpg'
        }
      }],
      currentClub: {
        books: []
      },
      addBookToClub: jest.fn()
    };
    wrapper = shallow(<BooksDisplay {...mockprops} />);

    const author = wrapper.find('.bookPreview').find('h5');

    expect(author.text()).toEqual('Author: jeff');
  });
});
