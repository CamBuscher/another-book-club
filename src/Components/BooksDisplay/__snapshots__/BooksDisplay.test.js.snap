// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksDisplay should match snapshot when given books 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addBookToClub={[Function]}
    books={
        Array [
            Object {
              "authors": Array [
                "Jeff",
              ],
              "imageLinks": Object {
                "thumbnail": "book.jpg",
              },
              "title": "Annihilation",
            },
          ]
    }
    currentClub={
        Object {
            "books": Array [],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="bookPreview"
>
          <h4>
                    Annihilation
          </h4>
          <h5>
                    Author: 
                    Jeff
          </h5>
          <img
                    className="bookPreviewImg"
                    src="book.jpg"
          />
          <br />
          <button
                    onClick={[Function]}
          >
                     Add to your club!
          </button>
</div>,
      ],
      "className": "bookshelf",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Annihilation
</h4>,
            <h5>
              Author: 
              Jeff
</h5>,
            <img
              className="bookPreviewImg"
              src="book.jpg"
/>,
            <br />,
            <button
              onClick={[Function]}
>
               Add to your club!
</button>,
          ],
          "className": "bookPreview",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Annihilation",
            },
            "ref": null,
            "rendered": "Annihilation",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Author: ",
                "Jeff",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Author: ",
              "Jeff",
            ],
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bookPreviewImg",
              "src": "book.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Add to your club!",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": " Add to your club!",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="bookPreview"
>
            <h4>
                        Annihilation
            </h4>
            <h5>
                        Author: 
                        Jeff
            </h5>
            <img
                        className="bookPreviewImg"
                        src="book.jpg"
            />
            <br />
            <button
                        onClick={[Function]}
            >
                         Add to your club!
            </button>
</div>,
        ],
        "className": "bookshelf",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Annihilation
</h4>,
              <h5>
                Author: 
                Jeff
</h5>,
              <img
                className="bookPreviewImg"
                src="book.jpg"
/>,
              <br />,
              <button
                onClick={[Function]}
>
                 Add to your club!
</button>,
            ],
            "className": "bookPreview",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Annihilation",
              },
              "ref": null,
              "rendered": "Annihilation",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Author: ",
                  "Jeff",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Author: ",
                "Jeff",
              ],
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "bookPreviewImg",
                "src": "book.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Add to your club!",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": " Add to your club!",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`BooksDisplay should match snapshot when given no books 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addBookToClub={[Function]}
    currentClub={
        Object {
            "books": Array [],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "bookshelf",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "bookshelf",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
